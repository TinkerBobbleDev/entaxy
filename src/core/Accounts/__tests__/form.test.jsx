import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { AccountFormComponent } from '../form'
import institutions from '../../../data/institutions'

describe('Account form', () => {
  const mochHandleSubmit = jest.fn()
  const mochHandleChange = jest.fn()
  const mochSetFieldValue = jest.fn()
  const mochHandleDelete = jest.fn()
  const mochHandleCancel = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('matches snapshot for new account', () => {
    const component = renderer.create((
      <AccountFormComponent
        institutions={[{ label: 'TD', value: 'TD' }]}
        handleSubmit={mochHandleSubmit}
        values={{}}
        errors={{}}
        touched={{}}
        isSubmitting={false}
        handleChange={mochHandleChange}
        setFieldValue={mochSetFieldValue}
        handleDelete={mochHandleDelete}
        handleCancel={mochHandleCancel}
        classes={{ }}
        accountInstitutions={[]}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('matches snapshot for edit account', () => {
    const component = renderer.create((
      <AccountFormComponent
        institutions={[{ label: 'TD', value: 'TD' }]}
        account={{ id: 1 }}
        handleSubmit={mochHandleSubmit}
        values={{}}
        errors={{}}
        touched={{}}
        isSubmitting={false}
        handleChange={mochHandleChange}
        setFieldValue={mochSetFieldValue}
        handleDelete={mochHandleDelete}
        handleCancel={mochHandleCancel}
        classes={{ }}
        accountInstitutions={['My institution']}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  describe('Component methods', () => {
    const wrapper = shallow((
      <AccountFormComponent
        institutions={[{ label: 'TD', value: 'TD' }]}
        account={{ id: 1 }}
        handleSubmit={mochHandleSubmit}
        values={{}}
        errors={{}}
        touched={{}}
        isSubmitting={false}
        handleChange={mochHandleChange}
        setFieldValue={mochSetFieldValue}
        handleDelete={mochHandleDelete}
        handleCancel={mochHandleCancel}
        classes={{ }}
        accountInstitutions={['My institution']}
      />
    ))
    const instance = wrapper.instance()

    it('should handleInstitutionChange', () => {
      instance.setState({ hideInstitutionOptions: true })
      expect(instance.state.hideInstitutionOptions).toBeTruthy()

      instance.handleInstitutionChange([
        'institution',
        { label: 'Bank of Montreal', value: 'BMO' }
      ])
      expect(mochSetFieldValue).toHaveBeenCalled()
      expect(instance.state.hideInstitutionOptions).toBeFalsy()
    })

    it('should handleCloseInstitutionOptions', () => {
      instance.setState({ hideInstitutionOptions: false })
      expect(instance.state.hideInstitutionOptions).toBeFalsy()

      instance.handleCloseInstitutionOptions()
      expect(instance.state.hideInstitutionOptions).toBeTruthy()
    })

    it('should display institutionOptions only if institution has API', () => {
      instance.setState({ hideInstitutionOptions: false })
      expect(instance.state.hideInstitutionOptions).toBeFalsy()

      expect(instance.institutionOptions()).toBeNull()

      expect(institutions['Bank of Montreal'].importTypes).not.toContain('API')
      expect(instance.institutionOptions({ value: 'Bank of Montreal' })).toBeNull()

      expect(institutions.Coinbase.importTypes).toContain('API')
      expect(instance.institutionOptions({ value: 'Coinbase' })).not.toBeNull()
    })

    it('should never display institutionOptions if hideInstitutionOptions is true', () => {
      instance.setState({ hideInstitutionOptions: true })
      expect(instance.state.hideInstitutionOptions).toBeTruthy()

      expect(instance.institutionOptions()).toBeNull()

      expect(institutions['Bank of Montreal'].importTypes).not.toContain('API')
      expect(instance.institutionOptions({ value: 'Bank of Montreal' })).toBeNull()

      expect(institutions.Coinbase.importTypes).toContain('API')
      expect(instance.institutionOptions({ value: 'Coinbase' })).toBeNull()
    })
  })
})
