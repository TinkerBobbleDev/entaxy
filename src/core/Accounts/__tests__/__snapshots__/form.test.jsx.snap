// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account form matches snapshot for edit account 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div>
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        Edit account
      </h6>
      <button
        aria-label="Close"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
      </button>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <form
      onSubmit={[MockFunction]}
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div>
            <span
              className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
            >
              Account type
            </span>
            <div
              className=" css-2b097c-container"
              onKeyDown={[Function]}
            >
              <div
                className=" css-yk16xz-control"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <div
                  className=" css-1hwfws3"
                >
                  <div
                    className=" css-1uccc91-singleValue"
                  >
                    Bank
                  </div>
                  <div
                    className="css-w8afj7-Input"
                  >
                    <div
                      className=""
                      style={
                        Object {
                          "display": "inline-block",
                        }
                      }
                    >
                      <input
                        aria-autocomplete="list"
                        autoCapitalize="none"
                        autoComplete="off"
                        autoCorrect="off"
                        disabled={false}
                        id="react-select-5-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        spellCheck="false"
                        style={
                          Object {
                            "background": 0,
                            "border": 0,
                            "boxSizing": "content-box",
                            "color": "inherit",
                            "fontSize": "inherit",
                            "label": "input",
                            "opacity": 1,
                            "outline": 0,
                            "padding": 0,
                            "width": "1px",
                          }
                        }
                        tabIndex="0"
                        type="text"
                        value=""
                      />
                      <div
                        style={
                          Object {
                            "height": 0,
                            "left": 0,
                            "overflow": "scroll",
                            "position": "absolute",
                            "top": 0,
                            "visibility": "hidden",
                            "whiteSpace": "pre",
                          }
                        }
                      >
                        
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className=" css-1g48xl4-IndicatorsContainer"
                >
                  <span
                    className=" css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    className=" css-tlfecz-indicatorContainer"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                  >
                    <svg
                      aria-hidden="true"
                      className="css-tj5bde-Svg"
                      focusable="false"
                      height={20}
                      viewBox="0 0 20 20"
                      width={20}
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <input
                name="accountType"
                type="hidden"
                value="bank"
              />
            </div>
          </div>
          <div>
            <span
              className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
            >
              Institution
            </span>
            <div
              className=" css-2b097c-container"
              onKeyDown={[Function]}
            >
              <div
                className=" css-yk16xz-control"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <div
                  className=" css-1hwfws3"
                >
                  <div
                    className=" css-1wa3eu0-placeholder"
                  >
                    Select or type to create ...
                  </div>
                  <div
                    className="css-w8afj7-Input"
                  >
                    <div
                      className=""
                      style={
                        Object {
                          "display": "inline-block",
                        }
                      }
                    >
                      <input
                        aria-autocomplete="list"
                        autoCapitalize="none"
                        autoComplete="off"
                        autoCorrect="off"
                        disabled={false}
                        id="react-select-6-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        spellCheck="false"
                        style={
                          Object {
                            "background": 0,
                            "border": 0,
                            "boxSizing": "content-box",
                            "color": "inherit",
                            "fontSize": "inherit",
                            "label": "input",
                            "opacity": 1,
                            "outline": 0,
                            "padding": 0,
                            "width": "1px",
                          }
                        }
                        tabIndex="0"
                        type="text"
                        value=""
                      />
                      <div
                        style={
                          Object {
                            "height": 0,
                            "left": 0,
                            "overflow": "scroll",
                            "position": "absolute",
                            "top": 0,
                            "visibility": "hidden",
                            "whiteSpace": "pre",
                          }
                        }
                      >
                        
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className=" css-1g48xl4-IndicatorsContainer"
                >
                  <span
                    className=" css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    className=" css-tlfecz-indicatorContainer"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                  >
                    <svg
                      aria-hidden="true"
                      className="css-tj5bde-Svg"
                      focusable="false"
                      height={20}
                      viewBox="0 0 20 20"
                      width={20}
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <input
                name="institution"
                type="hidden"
                value=""
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
            >
              Account name
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                aria-label="Account name"
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                maxLength={40}
                name="name"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
            >
              Opening balance
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                aria-label="Opening balance"
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                name="openingBalance"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                step={0.01}
                type="number"
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              aria-label="Opening balance date"
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
              data-shrink={true}
            >
              Opening balance date
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                name="openingBalanceDate"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="date"
              />
            </div>
          </div>
          <div>
            <span
              className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
            >
              Account currency
            </span>
            <div
              className=" css-2b097c-container"
              onKeyDown={[Function]}
            >
              <div
                className=" css-yk16xz-control"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <div
                  className=" css-1hwfws3"
                >
                  <div
                    className=" css-1wa3eu0-placeholder"
                  >
                    Select...
                  </div>
                  <div
                    className="css-w8afj7-Input"
                  >
                    <div
                      className=""
                      style={
                        Object {
                          "display": "inline-block",
                        }
                      }
                    >
                      <input
                        aria-autocomplete="list"
                        autoCapitalize="none"
                        autoComplete="off"
                        autoCorrect="off"
                        disabled={false}
                        id="react-select-7-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        spellCheck="false"
                        style={
                          Object {
                            "background": 0,
                            "border": 0,
                            "boxSizing": "content-box",
                            "color": "inherit",
                            "fontSize": "inherit",
                            "label": "input",
                            "opacity": 1,
                            "outline": 0,
                            "padding": 0,
                            "width": "1px",
                          }
                        }
                        tabIndex="0"
                        type="text"
                        value=""
                      />
                      <div
                        style={
                          Object {
                            "height": 0,
                            "left": 0,
                            "overflow": "scroll",
                            "position": "absolute",
                            "top": 0,
                            "visibility": "hidden",
                            "whiteSpace": "pre",
                          }
                        }
                      >
                        
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className=" css-1g48xl4-IndicatorsContainer"
                >
                  <div
                    aria-hidden="true"
                    className=" css-at12u2-loadingIndicator"
                  >
                    <span
                      className="css-1ojr9sz-LoadingDot"
                    />
                    <span
                      className="css-12a6fd1-LoadingDot"
                    />
                    <span
                      className="css-19d8ryt-LoadingDot"
                    />
                  </div>
                  <span
                    className=" css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    className=" css-tlfecz-indicatorContainer"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                  >
                    <svg
                      aria-hidden="true"
                      className="css-tj5bde-Svg"
                      focusable="false"
                      height={20}
                      viewBox="0 0 20 20"
                      width={20}
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <input
                name="currency"
                type="hidden"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <hr
        className="MuiDivider-root"
      />
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Delete this account
          </span>
        </button>
        <div />
        <div
          className="SubmitButtonWithProgress-wrapper-272"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Save
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`Account form matches snapshot for new account 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div>
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        New account
      </h6>
      <button
        aria-label="Close"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
      </button>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <form
      onSubmit={[MockFunction]}
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div>
            <span
              className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
            >
              Account type
            </span>
            <div
              className=" css-2b097c-container"
              onKeyDown={[Function]}
            >
              <div
                className=" css-yk16xz-control"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <div
                  className=" css-1hwfws3"
                >
                  <div
                    className=" css-1uccc91-singleValue"
                  >
                    Bank
                  </div>
                  <div
                    className="css-w8afj7-Input"
                  >
                    <div
                      className=""
                      style={
                        Object {
                          "display": "inline-block",
                        }
                      }
                    >
                      <input
                        aria-autocomplete="list"
                        autoCapitalize="none"
                        autoComplete="off"
                        autoCorrect="off"
                        disabled={false}
                        id="react-select-2-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        spellCheck="false"
                        style={
                          Object {
                            "background": 0,
                            "border": 0,
                            "boxSizing": "content-box",
                            "color": "inherit",
                            "fontSize": "inherit",
                            "label": "input",
                            "opacity": 1,
                            "outline": 0,
                            "padding": 0,
                            "width": "1px",
                          }
                        }
                        tabIndex="0"
                        type="text"
                        value=""
                      />
                      <div
                        style={
                          Object {
                            "height": 0,
                            "left": 0,
                            "overflow": "scroll",
                            "position": "absolute",
                            "top": 0,
                            "visibility": "hidden",
                            "whiteSpace": "pre",
                          }
                        }
                      >
                        
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className=" css-1g48xl4-IndicatorsContainer"
                >
                  <span
                    className=" css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    className=" css-tlfecz-indicatorContainer"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                  >
                    <svg
                      aria-hidden="true"
                      className="css-tj5bde-Svg"
                      focusable="false"
                      height={20}
                      viewBox="0 0 20 20"
                      width={20}
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <input
                name="accountType"
                type="hidden"
                value="bank"
              />
            </div>
          </div>
          <div>
            <span
              className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
            >
              Institution
            </span>
            <div
              className=" css-2b097c-container"
              onKeyDown={[Function]}
            >
              <div
                className=" css-yk16xz-control"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <div
                  className=" css-1hwfws3"
                >
                  <div
                    className=" css-1wa3eu0-placeholder"
                  >
                    Select or type to create ...
                  </div>
                  <div
                    className="css-w8afj7-Input"
                  >
                    <div
                      className=""
                      style={
                        Object {
                          "display": "inline-block",
                        }
                      }
                    >
                      <input
                        aria-autocomplete="list"
                        autoCapitalize="none"
                        autoComplete="off"
                        autoCorrect="off"
                        disabled={false}
                        id="react-select-3-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        spellCheck="false"
                        style={
                          Object {
                            "background": 0,
                            "border": 0,
                            "boxSizing": "content-box",
                            "color": "inherit",
                            "fontSize": "inherit",
                            "label": "input",
                            "opacity": 1,
                            "outline": 0,
                            "padding": 0,
                            "width": "1px",
                          }
                        }
                        tabIndex="0"
                        type="text"
                        value=""
                      />
                      <div
                        style={
                          Object {
                            "height": 0,
                            "left": 0,
                            "overflow": "scroll",
                            "position": "absolute",
                            "top": 0,
                            "visibility": "hidden",
                            "whiteSpace": "pre",
                          }
                        }
                      >
                        
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className=" css-1g48xl4-IndicatorsContainer"
                >
                  <span
                    className=" css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    className=" css-tlfecz-indicatorContainer"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                  >
                    <svg
                      aria-hidden="true"
                      className="css-tj5bde-Svg"
                      focusable="false"
                      height={20}
                      viewBox="0 0 20 20"
                      width={20}
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <input
                name="institution"
                type="hidden"
                value=""
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
            >
              Account name
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                aria-label="Account name"
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                maxLength={40}
                name="name"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
            >
              Opening balance
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                aria-label="Opening balance"
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                name="openingBalance"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                step={0.01}
                type="number"
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              aria-label="Opening balance date"
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
              data-shrink={true}
            >
              Opening balance date
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                name="openingBalanceDate"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="date"
              />
            </div>
          </div>
          <div>
            <span
              className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
            >
              Account currency
            </span>
            <div
              className=" css-2b097c-container"
              onKeyDown={[Function]}
            >
              <div
                className=" css-yk16xz-control"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <div
                  className=" css-1hwfws3"
                >
                  <div
                    className=" css-1wa3eu0-placeholder"
                  >
                    Select...
                  </div>
                  <div
                    className="css-w8afj7-Input"
                  >
                    <div
                      className=""
                      style={
                        Object {
                          "display": "inline-block",
                        }
                      }
                    >
                      <input
                        aria-autocomplete="list"
                        autoCapitalize="none"
                        autoComplete="off"
                        autoCorrect="off"
                        disabled={false}
                        id="react-select-4-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        spellCheck="false"
                        style={
                          Object {
                            "background": 0,
                            "border": 0,
                            "boxSizing": "content-box",
                            "color": "inherit",
                            "fontSize": "inherit",
                            "label": "input",
                            "opacity": 1,
                            "outline": 0,
                            "padding": 0,
                            "width": "1px",
                          }
                        }
                        tabIndex="0"
                        type="text"
                        value=""
                      />
                      <div
                        style={
                          Object {
                            "height": 0,
                            "left": 0,
                            "overflow": "scroll",
                            "position": "absolute",
                            "top": 0,
                            "visibility": "hidden",
                            "whiteSpace": "pre",
                          }
                        }
                      >
                        
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className=" css-1g48xl4-IndicatorsContainer"
                >
                  <div
                    aria-hidden="true"
                    className=" css-at12u2-loadingIndicator"
                  >
                    <span
                      className="css-1ojr9sz-LoadingDot"
                    />
                    <span
                      className="css-12a6fd1-LoadingDot"
                    />
                    <span
                      className="css-19d8ryt-LoadingDot"
                    />
                  </div>
                  <span
                    className=" css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    className=" css-tlfecz-indicatorContainer"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                  >
                    <svg
                      aria-hidden="true"
                      className="css-tj5bde-Svg"
                      focusable="false"
                      height={20}
                      viewBox="0 0 20 20"
                      width={20}
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <input
                name="currency"
                type="hidden"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <hr
        className="MuiDivider-root"
      />
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Delete this account
          </span>
        </button>
        <div />
        <div
          className="SubmitButtonWithProgress-wrapper-272"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Save
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
